compiler/pipes/calc-real-defines-values.cpp

```diff
@@ -25,6 +27,18 @@ void CalcRealDefinesValuesF::on_finish(DataStream<FunctionPtr> &os) {
   }
 
   stage::die_if_global_errors();
+
+  // after all constants have been inlined, call RegisterKphpConfiguration
+  // do this INSIDE this sync point, BEFORE passing all functions to the next pipe
+  // this is done intentionally, so that further phpdoc parsing has the KphpConfiguration registered
+  {
+    ClassPtr kphpConfigurationClass = G->get_class("KphpConfiguration");
```

Comment:
may be 
`SyncC<RegisterKphpConfiguration>{} ` right after `SyncC<CalcRealDefinesValuesF>{}`
with logic in `RegisterKphpConfiguration::execute()` and simple `on_finish` would be better?

I don't like this hack, it looks wired, because in this case `RegisterKphpConfiguration` is not a pipe anymore and should be moved into another place.

Reply:
Okay, I don't like this moment also. But two sync pipes one after one seemed strange for me also :)
How'd you suggest to make this finally?
1) Two sync pipes
2) make RegisterKphpConfiguration not a pipe, but a standalone logic
?

Reply:
Okay, I don't like this moment also. But two sync pipes one after one seemed strange for me also :)
How'd you suggest to make this finally?
1) Two sync pipes
2) make RegisterKphpConfiguration not a pipe, but a standalone logic
?

Reply:
From my point of view two sync pipes is the easiest way, and I believe it wouldn't make any performance penalty. An in additional, if we move it into another place, then the `CalcRealDefinesValuesF` pipe should be renamed as well to something like
`CalcRealDefinesValuesAndRegisterConfigurationF`. Therefore, yes, I vote for two sync pipes.

Reply:
Ok, I made two sync pipes. Maybe, we'll discuss this approach later :)